realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/ProgressListenerTests.kt

```diff
@@ -147,171 +141,198 @@ class ProgressListenerTests {
         }
-        TestHelper.awaitOrFail(allChangeUploaded)
-        realm.close()
     }
 
     @Test
     fun addListenerInsideCallback_mixProgressModes() {
         val allChangeUploaded = CountDownLatch(3)
         val progressCompletedReported = AtomicBoolean(false)
         val config = createSyncConfig()
-        val realm = Realm.getInstance(config)
-        writeSampleData(realm)
-        val session: SyncSession = realm.syncSession
-        session.addUploadProgressListener(ProgressMode.INDEFINITELY) { progress ->
-            if (progress.isTransferComplete) {
-                allChangeUploaded.countDown()
-                if (progressCompletedReported.compareAndSet(false, true)) {
-                    val realm = Realm.getInstance(config)
-                    writeSampleData(realm)
-                    realm.close()
-                    session.addUploadProgressListener(ProgressMode.CURRENT_CHANGES) { progress ->
-                        if (progress.isTransferComplete) {
-                            allChangeUploaded.countDown()
+        Realm.getInstance(config).use { realm ->
+            val session: SyncSession = realm.syncSession
+            session.addUploadProgressListener(ProgressMode.INDEFINITELY) { progress ->
+                if (progress.isTransferComplete) {
+                    allChangeUploaded.countDown()
+                    if (progressCompletedReported.compareAndSet(false, true)) {
+                        Realm.getInstance(config).use { realm ->
+                            writeSampleData(realm)
+                        }
+                        session.addUploadProgressListener(ProgressMode.CURRENT_CHANGES) { progress ->
+                            if (progress.isTransferComplete) {
+                                allChangeUploaded.countDown()
+                            }
                         }
                     }
                 }
             }
+            writeSampleData(realm)
+            TestHelper.awaitOrFail(allChangeUploaded)
         }
-        TestHelper.awaitOrFail(allChangeUploaded)
-        realm.close()
     }
 
     @Test
     fun addProgressListener_triggerImmediatelyWhenRegistered() {
         val config = createSyncConfig()
-        val realm = Realm.getInstance(config)
-        val session: SyncSession = realm.syncSession
-        checkListener(session, ProgressMode.INDEFINITELY)
-        checkListener(session, ProgressMode.CURRENT_CHANGES)
-        realm.close()
+        Realm.getInstance(config).use { realm ->
+            val session: SyncSession = realm.syncSession
+            checkDownloadListener(session, ProgressMode.INDEFINITELY)
+            checkUploadListener(session, ProgressMode.INDEFINITELY)
+            checkDownloadListener(session, ProgressMode.CURRENT_CHANGES)
+            checkUploadListener(session, ProgressMode.CURRENT_CHANGES)
+        }
+    }
+
+    @Test
+    @Ignore("Tracked by https://github.com/realm/realm-java/issues/6976")
```

Comment:
```suggestion
    @Ignore("FIXME: Tracked by https://github.com/realm/realm-java/issues/6976")
```

Makes it easier to find again when searching the codebase